packages/@lwc/template-compiler/src/parser/attribute.ts

```diff
@@ -292,9 +292,15 @@ export function attributeToPropertyName(element: IRElement, attrName: string): s
     let propName = attrName;
     if (shouldCamelCaseAttribute(element, attrName)) {
         const attrToSplit = ATTRS_PROPS_TRANFORMS[propName] || propName;
-        propName = attrToSplit
+
+        // flag indicating whether leading character should be uppercased
+        const pascalCase = attrToSplit.startsWith('-');
+
+        // remove leading hyphen
+        const normalizedAttrToSplit = pascalCase ? attrToSplit.replace(/^[-]+/, '') : attrToSplit;
+        propName = normalizedAttrToSplit
             .split('_')
-            .map(camelcase)
+            .map(input => camelcase(input, { pascalCase }))
```

Comment:
I believe this will return `Foo_Bar` for `foo_bar` when previously it used to return `foo_bar` and that doesn't look correct. Either way, can we add a test case for it?

Reply:
the pascal will only be set if your attribute name starts with hyphen. Your example should work as expected

Reply:
lets add a test to be sure :)

Reply:
Please add a test for the attribute `-foo_bar_baz`

Reply:
@ekashida great catch! Pascal case will convert every leading character to uppercase. Just ran the test and the result was: Foo_Bar_Baz